### PR TITLE
mm-spinner and mm-loader port

### PR DESCRIPTION
**Pull request by shuwen**

### CODE REVIEW

**Comment by dlasky** [comment]
@derekolson can you look at this as well since you did the .5x version of them. thx

File: src/mm-loader/mm-loader.js [hunk 1]
```diff
@@ -4,39 +4,72 @@
...
+			this.style.display = "block";
+			if(this.spinner) this.spinner.start();
+		},
+
+		hide: function () {
+			this.style.display = "none";
+			if(this.spinner) this.spinner.stop();
+		},
+
+		_updateStyle: function() {
```
Comment (anthonykoerber) [inline src/mm-loader/mm-loader.js]: should be something like: 
````
_updateStyle: function(spinnerRadius, bgColor, bgOpacity, paddingTop, paddingLeft) {
			return this.styleBlock({
				background: this._convertHex(bgColor),
				borderRadius: spinnerRadius*2 + "px",
				paddingTop: paddingTop + "px",
				paddingRight: paddingLeft + "px",
				paddingBottom: paddingTop + "px",
				paddingLeft: paddingLeft + "px",
			});
		},
```

File: src/mm-loader/mm-loader.html [hunk 2]
```diff
@@ -5,20 +5,22 @@
...
+<dom-module id="mm-loader">
+
+	<link href="mm-loader.css" type="css" rel="import"/>
+
 	<template>
-		
-		<link href="mm-loader.css" rel="stylesheet" type="text/css"/>
-		
-		<div class="_mm_loader" style="background:{{ bgColor | convertHex }}; border-radius:{{ spinner.radius*2 }}px; padding:{{ paddingTop }}px {{paddingLeft}}px {{paddingTop}}px {{paddingLeft}}px;">
+		<div class="_mm_loader" style$="{{_updateStyle}}">
```
Comment (anthonykoerber) [inline src/mm-loader/mm-loader.html]: Should be like: 
````
<div class="_mm_loader" style$="{{ _updateStyle(spinnerRadius, bgColor, bgOpacity, paddingTop, paddingLeft) }}">
```

**Comment by anthonykoerber** [comment]
+ 1 on spinner, loader needs a change to display it's background properly, but otherwise is a +1

**Comment by shuwen** [comment]
yeah there was a problem with the timing when using shadow dom, fixed in #21

**Pull request merged**